Makefile

```diff
@@ -0,0 +1,73 @@
+# Tool versions
+TERRAFORM_VERSION := 1.8.0
+TFLINT_VERSION := 0.51.1
+ACTIONLINT_VERSION := 1.7.7
+
+# OS and architecture detection
+OS := $(shell uname -s | tr '[:upper:]' '[:lower:]')
+ARCH := $(shell uname -m | sed 's/x86_64/amd64/; s/arm64/arm64/')
+
+VENV := .venv
+
+# Terraform variables
+TF_WORKSPACE ?= default
+TF_DIR := terraform
+
+.PHONY: deps
+deps:
+	@mkdir -p ${VENV}/bin
+
+.PHONY: tflint-install
+tflint-install: deps
+	@if ! command -v ${VENV}/bin/tflint >/dev/null 2>&1 || [ "$$(${VENV}/bin/tflint --version | head -1 | grep -Eo '[0-9]+\.[0-9]+\.[0-9]+')" != "${TFLINT_VERSION}" ]; then \
+		echo "Installing tflint v${TFLINT_VERSION}..."; \
+		echo "Downloading for OS: ${OS}, ARCH: ${ARCH}"; \
+		mkdir -p ${VENV}/bin; \
+		cd ${VENV}/bin; \
+		curl -sL "https://github.com/terraform-linters/tflint/releases/download/v${TFLINT_VERSION}/tflint_${OS}_${ARCH}.zip" -o tflint.zip; \
+		unzip -o tflint.zip; \
+		rm tflint.zip; \
+		chmod +x tflint; \
+		cd -; \
```

Comment:
As an alternative to using `cd foo; cd -`, you could consider doing `pushd foo; popd`.  This is the more unix-blessed way of doing things, though of course `cd -` is fine too, and will work in any shell we're likely to use.